pay-api/src/pay_api/services/eft_short_name_historical.py

```diff
@@ -0,0 +1,194 @@
+
+    @classmethod
+    def search(cls, short_name_id: int,
+               search_criteria: EFTShortnameHistorySearch = EFTShortnameHistorySearch()):
+        """Return EFT Short name history by search criteria."""
+        history_model = aliased(EFTShortnamesHistoricalModel)
+        latest_history_model = aliased(EFTShortnamesHistoricalModel)
+
+        latest_history_subquery = (
+            select(
+                latest_history_model.statement_number,
+                latest_history_model.transaction_type,
+                latest_history_model.is_processing
+            )
+            .where(
+                latest_history_model.short_name_id == history_model.short_name_id,
+                latest_history_model.statement_number == history_model.statement_number
+            )
+            .order_by(
+                latest_history_model.statement_number,
+                latest_history_model.transaction_date.desc(),
+                latest_history_model.id.desc()
+            )
+            .limit(1)
+            .correlate(history_model)
+        ).subquery('latest_statement_history')
+
+        reversible_statement_subquery = exists(
+            select(1)
+            .select_from(latest_history_subquery)
+            .where(and_(latest_history_subquery.c.transaction_type == EFTHistoricalTypes.STATEMENT_PAID.value,
+                        latest_history_subquery.c.is_processing == false()))
+        )
+
+        is_reversible_statement = case(
+            (
+                and_(
+                    history_model.transaction_type == EFTHistoricalTypes.STATEMENT_PAID.value,
+                    history_model.transaction_date >= cls.transaction_date_now() - timedelta(days=60)
+                ),
+                reversible_statement_subquery
+            ),
+            else_=False
+        )
+
+        query = (db.session.query(history_model.id,
+                                  history_model.short_name_id,
+                                  history_model.amount,
+                                  history_model.credit_balance,
+                                  history_model.description,
+                                  history_model.statement_number,
+                                  history_model.transaction_date,
+                                  history_model.transaction_type,
+                                  history_model.is_processing,
+                                  PaymentAccountModel.auth_account_id,
+                                  cls._get_account_name(),
+                                  PaymentAccountModel.branch_name.label('account_branch'),
+                                  is_reversible_statement.label('is_reversible'))
+                 .outerjoin(PaymentAccountModel, PaymentAccountModel.id == history_model.payment_account_id)
+                 .filter(history_model.short_name_id == short_name_id)
+                 .filter(history_model.hidden == false()))
```

Comment:
interesting,  I usually see `history_model.hidden.is_(False)`

Reply:
updated to be more consistent with what is currently used